src/api/ledger/pathfind.js

```diff
@@ -1,147 +1,106 @@
+  });
+}
 
-  function prepareOptions() {
-    const pathfindParams = {
-      src_account: pathfind.source.address,
-      dst_account: pathfind.destination.address,
-      dst_amount: utils.common.toRippledAmount(pathfind.destination.amount)
-    };
-    if (typeof pathfindParams.dst_amount === 'object'
-          && !pathfindParams.dst_amount.issuer) {
-      // Convert blank issuer to sender's address
-      // (Ripple convention for 'any issuer')
-      // https://ripple.com/build/transactions/
-      //     #special-issuer-values-for-sendmax-and-amount
-      // https://ripple.com/build/ripple-rest/#counterparties-in-payments
-      pathfindParams.dst_amount.issuer = pathfindParams.dst_account;
-    }
-    if (pathfind.source.amounts && pathfind.source.amounts.length > 0) {
-      pathfindParams.src_currencies = pathfind.source.amounts;
-    }
-    return pathfindParams;
+function requestPathFind(remote, pathfind, callback) {
+  const params = {
+    src_account: pathfind.source.address,
+    dst_account: pathfind.destination.address,
+    dst_amount: utils.common.toRippledAmount(pathfind.destination.amount)
+  };
+  if (typeof params.dst_amount === 'object' && !params.dst_amount.issuer) {
+    // Convert blank issuer to sender's address
+    // (Ripple convention for 'any issuer')
+    // https://ripple.com/build/transactions/
+    //     #special-issuer-values-for-sendmax-and-amount
+    // https://ripple.com/build/ripple-rest/#counterparties-in-payments
+    params.dst_amount.issuer = params.dst_account;
+  }
+  if (pathfind.source.amounts && pathfind.source.amounts.length > 0) {
+    params.src_currencies = pathfind.source.amounts.map(amount =>
+      _.omit(amount, 'value'));
   }
 
-  function findPath(pathfindParams, _callback) {
-    const request = self.remote.requestRipplePathFind(pathfindParams);
-    request.once('error', _callback);
-    request.once('success', function(pathfindResults) {
-      pathfindResults.source_account = pathfindParams.src_account;
-      pathfindResults.source_currencies = pathfindParams.src_currencies;
-      pathfindResults.destination_amount = pathfindParams.dst_amount;
-      _callback(null, pathfindResults);
-    });
+  remote.requestRipplePathFind(params,
+    composeAsync(_.partial(addParams, params), callback));
+}
 
-    function reconnectRippled() {
-      self.remote.disconnect(function() {
-        self.remote.connect();
-      });
-    }
-    request.timeout(utils.common.server.CONNECTION_TIMEOUT, function() {
-      request.removeAllListeners();
-      reconnectRippled();
-      _callback(new TimeOutError('Path request timeout'));
+function addDirectXrpPath(paths, xrpBalance) {
+  // Add XRP "path" only if the source acct has enough XRP to make the payment
+  const destinationAmount = paths.destination_amount;
+  if ((new BigNumber(xrpBalance)).greaterThanOrEqualTo(destinationAmount)) {
+    paths.alternatives.unshift({
+      paths_canonical: [],
```

Comment:
`paths_canonical` is basically never used afaik. I'm not sure if it's officially deprecated, but what matters is what fields the ripple-api formatter requires. `parsePathFind` uses only `paths_computed`.